### DISCUSSION TITLE
Worth updating methods signatures from T to Target?

### DISCUSSION THREAD

**Discussion by roman-mibex-2** [com-lihaoyi/mill#3968] [Development]
When I auto complete to override an existing target, the the IDE (IntelliJ) will use `T[ReturnType]`.

However, these are aliases to full named types since 0.12 and those are used in the introduction chapters. 
So, my question is it worth to update these signatures so that auto complete gives types matching the docs? Or will that cause some strange issue and and thats why it was kept as the `T` alias?

If it doesn't cause issues and its something worth doing then I can certainly do some of that busy work.

Sub questions:
- Is 'Target' or 'Task' preferred. Both exist?

**Comment by roman-mibex-2** [comment]
Oh, it probably breaks compatability with 0.11 plugins, because those refer to `mill.T`? I'm not familiar enough with Scala API comparability.

**Comment by lihaoyi** [comment]
Currently the recommended type signature is `T[_]`. For now it is an alias for `Target[_]`, but I expect in 0.13.0 to make it an alias for `Task[_]`, but using `T[_]` is probably the most future proof

**Comment by roman-mibex-2** [comment]
Conclusion: No renaming for now to be on the safe side.
At least for 0.12.x.
For 0.13.x maybe more possible, but maybe adds more churn than helping.
